src/utils.py

```diff
@@ -0,0 +1,85 @@
+"""Utils"""
+import os
+from pathlib import Path
+from typing import Set, Union
+
+import pandas as pd
+import yaml
+
+
+def add_prefixes_to_plain_id(x: str) -> str:
+    """From plain IDs from originanl source, add prefixes.
+
+    Terms:
+        CN: stands for "CUI Novel". These are created for any MedGen records without UMLS CUI.
+        C: stands for "CUI". These are sourced from UMLS.
+        CUI: stands for "Concept Unique Identifier"
+        UID (Unique IDentifier): These are cases where the id is all digits; does not start with a leading alpha char.
+    """
+    return f'MEDGENCUI:{x}' if x.startswith('CN') \
+        else f'UMLS:{x}' if x.startswith('C') \
+        else f'MEDGEN:{x}'
+
+
+def find_prefixes_in_mapping_set(source_df: pd.DataFrame) -> Set[str]:
```

Comment:
## General: `find_prefixes_in_mapping_set()`
I'm using this just to improves cleanliness. I wanted a way to filter out any unused prefixes from the `curie_map` in the SSSOM TSV comment. I use this in `write_sssom()`, which is a new, small function that uses this and `write_tsv_with_comments()`.

@matentzn Is there a standard way in `sssom-py` for, when saving the TSV, to filter out any prefixes in the metadata which don't appear in any rows, as I've implemented here?

Reply:
MappingSetDataFrame() is what you want to use, it has clean_prefixes() or some such methods to do this.

Reply:
Done~. Refer to [related SSSOM comment](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1536735796) for further report.